src/bin/06_storage_iter.rs

```diff
@@ -0,0 +1,122 @@
+/*!
+This example iterates over the storage keys of the XcmPallet's VersionNotifiers,
+which is a double storage map.
+
+The example interprets the storage keys at the byte level and provides the
+output of the given keys.
+
+```
+cargo run --bin 06_storage_iter
+```
+ */
+
+use jsonrpsee::{core::client::ClientT, rpc_params};
+use parity_scale_codec::Decode;
+use sp_core::{hashing, storage::StorageKey};
+use utils::ws_client;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let client = ws_client("ws://localhost:9944").await?;
+
+    // The VersionNotifiers type of the XcmPallet is defined as:
+    //
+    // ```
+    //  All locations that we have requested version notifications from.
+    // 	#[pallet::storage]
+    // 	pub(super) type VersionNotifiers<T: Config> = StorageDoubleMap<
+    // 		_,
+    // 		Twox64Concat,
+    // 		XcmVersion,
+    // 		Blake2_128Concat,
+    // 		VersionedMultiLocation,
+    // 		QueryId,
+    // 		OptionQuery,
+    // 	>;
+    // ```
+
+    // Construct the XcmPallet VersionNotifiers key represented by:
+    // twox_128(XcmPallet) ++ twox_128(VersionNotifiers)
+    let storage_prefix = "XcmPallet";
+    let storage_name = "VersionNotifiers";
+
+    // Obtain the twox_128 of the provided values.
+    let storage_prefix_hashed = hashing::twox_128(storage_prefix.as_bytes());
+    let storage_name_hashed = hashing::twox_128(storage_name.as_bytes());
+
+    // Concat the values.
+    let mut storage_key = Vec::new();
+    storage_key.extend_from_slice(&storage_prefix_hashed);
+    storage_key.extend_from_slice(&storage_name_hashed);
+
+    // Hexify the above bytes and make the request to get the value back:
+    let storage_key_hex = format!("0x{}", hex::encode(&storage_key));
+    println!("VersionNotifiers storage key: {}", storage_key_hex);
+
+    let params = rpc_params![storage_key_hex.clone()];
+    let keys: Vec<StorageKey> = client.request("state_getKeys", params).await?;
+
+    println!("Obtained keys:");
+    for key in keys.iter() {
+        let key_hex = format!("0x{}", hex::encode(&key));
+        println!("Key: {}", key_hex);
+
+        // Obtain the byte representation of the key.
+        let key_bytes = &key.0;
+        // The first 32 bytes are identical to our storage query key and are represented by:
```

Comment:
This line can probably be removed now :)

Reply:
Done! Thanks